drivers/flash/flash_mspi_atxp032.c

```diff
@@ -745,7 +754,7 @@ static int flash_mspi_atxp032_read_jedec_id(const struct device *flash, uint8_t
 {
 	struct flash_mspi_atxp032_data *data = flash->data;
 
-	id = &data->jedec_id;
+	id = (uint8_t *)&data->jedec_id;
```

Comment:
I do not think this is correct, as the flash_read_jedec_id expects ID to be copied from what is read on device to buffer pointed by the id, but the buffer is specified by function as 3 byte (which is wrong anyway), and here it will overflow if ID is longer and also there is a question whether byte order is not changed on assignment?

Reply:
Yeah, your are right.